Content.Shared/Chat/ChatModifiers/ColorEntityNameHeaderChatModifier.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using Content.Shared.CCVar;
+using Content.Shared.Decals;
+using Robust.Shared.Configuration;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Chat.ChatModifiers;
+
+/// <summary>
+/// Wraps the [EntityNameHeader] tag in a [color] tag, should the player have colored names enabled.
+/// </summary>
+[Serializable]
+[DataDefinition]
+public sealed partial class ColorEntityNameHeaderChatModifier : ChatModifier
+{
+
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IConfigurationManager _config = default!;
+    private static ProtoId<ColorPalettePrototype> _chatNamePalette = "ChatNames";
+
+    public override FormattedMessage ProcessChatModifier(FormattedMessage message, Dictionary<Enum, object> channelParameters)
+    {
+        IoCManager.InjectDependencies(this);
+
+        var colorName = _config.GetCVar(CCVars.ChatEnableColorName);
+        if (colorName && message.TryFirstOrDefault(x => x.Name == "EntityNameHeader", out var nameHeader))
+        {
+            var name = nameHeader.Value.StringValue;
+            if (name != null)
+            {
+                var color = Color.TryFromHex(GetNameColor(name));
+                if (color != null)
```

Comment:
maybe reduce nesting by inverting conditions and early returning? for better readability :<